pydoctor/templatewriter/__init__.py

```diff
@@ -31,6 +32,15 @@ def parse_xml(text: str) -> minidom.Document:
     except Exception as e:
         raise ValueError(f"Failed to parse template as XML: {e}") from e
 
+def scandir(path: Path) -> Iterable['Template']:
+    """
+    Scan a directory for templates. 
+    """
+    for entry in os.scandir(path):
```

Comment:
`path.iterdir()` should work as well, then you don't have to convert the strings back into a `Path`s.

Reply:
It's not string it's os.DirEntry, but it does not have the suffix property so we have to cast it to Path

Reply:
Indeed it's a dir entry. But the only thing you use the dir entry for is to create the `Path` object, which you could get directly from `iterdir()` instead.